com.reprezen.swagedit/src/com/reprezen/swagedit/validation/SwaggerError.java

```diff
@@ -109,20 +128,66 @@ public String getMessage() {
 			builder.append("Failed to match exactly one schema:");
 			builder.append("\n");
 
-			for (String key: errors.keySet()) {
+			for (String location : orderedErrorLocations) {
 				builder.append(tabs);
 				builder.append(" - ");
-				builder.append(key);
+				builder.append(getHumanFriendlyText(location));
 				builder.append(":");
 				builder.append("\n");
 
-				for (SwaggerError e: errors.get(key)) {
+				for (SwaggerError e : errors.get(location)) {
 					builder.append(e.getMessage(true));
 				}
 			}
 
 			return builder.toString();
 		}
+
+		protected String getHumanFriendlyText(String location) {
+			JsonNode swaggerSchemaNode = findNode(location);
+			if (swaggerSchemaNode == null) {
+				return location;
+			}
+			JsonNode title = swaggerSchemaNode.get("title");
+			if (title != null) {
+				return title.asText();
+			}
+			JsonNode description = swaggerSchemaNode.get("title");
```

Comment:
I checked the Swagger Schema, and verified that all uses of the `oneOf` constraint use named schemas from `/definitions`.  So it should be safe to rely on this.

There are two uses of anyOf in "schema":

```
        "additionalProperties": {
          "anyOf": [
            {
              "$ref": "#/definitions/schema"
            },
            {
              "type": "boolean"
            }
          ],
```

and:
```
        "items": {
          "anyOf": [
            {
              "$ref": "#/definitions/schema"
            },
            {
              "type": "array",
              "minItems": 1,
              "items": {
                "$ref": "#/definitions/schema"
              }
            }
          ],
```

Also one more in "format":
```
    "format": {
      "anyOf": [
        {
          "title": "user-defined format",
          "type": "string"
        },
        {
          "title": "pre-defined format",
          "type": "string",
            "enum": [
              "int32",
              "int64",
              "float",
              "double",
              "byte",
              "binary",
              "date",
              "date-time",
              "password"          			
          	]
        }
      ]   
    }
```

I'm checking in a suggested change that adds titles to these cases.